runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -97,7 +97,13 @@ function TSHighlighter.new(tree, opts)
   if vim.g.syntax_on ~= 1 then
     vim.api.nvim_command('runtime! syntax/synload.vim')
   end
-  vim.bo[self.bufnr].spelloptions = 'noplainbuffer'
+
+  -- vim.cmd'set spelloptions+=noplainbuffer'
```

Comment:
`vim.opt.spelloptions:append('noplainbuffer')`

Reply:
The only reason I didn't go for that is that you can't pass a buffer handle to `vim.opt_local`.

Reply:
Let's try it out -- if it's needed, it should be added to `vim.opt_local`.